atoma-state/src/state_manager.rs

```diff
@@ -5380,12 +5382,55 @@ mod tests {
             stack_computed_units,
             is_confidential,
         } = state
-            .update_stack_num_compute_units(1, 200, 200, 0.95)
+            .update_stack_num_compute_units(1, 200, 200, 0.95, 0)
             .await?;
         assert_eq!((ratio * 100.0) as i64, 20); // No change: 200 / 1000 = 0.2
         assert_eq!(stack_computed_units, 200);
         assert!(is_confidential);
 
+        state.insert_new_stack(Stack {
+            stack_small_id: 2,
+            task_small_id: 1,
+            num_compute_units: 1000,
+            already_computed_units: 960,
+            owner_address: "0x1".to_string(),
+            stack_id: "0x2".to_string(),
+            selected_node_id: 1,
+            price_per_one_million_compute_units: 1000,
+            in_settle_period: false,
+            total_hash: vec![0; 32],
+            num_total_messages: 0,
+            is_claimed: false,
+            is_locked_for_claim: false,
+        }).await?;
+        // Test case 6: Edge case - estimated equals num_compute_units, but with concurrent requests
+        let UpdateStackNumComputeUnitsAndClaimFunds {
+            ratio,
+            stack_computed_units,
+            is_confidential,
+        } = state
+            .update_stack_num_compute_units(2, 50, 50, 0.95, 1)
```

Comment:
the `stack_small_id` here is first parameter, but in the SQL query it's bound to `$3`.  You can either re-order the bindings or modify the SQL query to match the current binding order as suggested [here](https://github.com/atoma-network/atoma-node/pull/541/files/98d1aec7c8dc86e3304d8a7b51ce3a8bfc32b7b2#r2029310325)

Reply:
Address the comment.